geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapDestroy.java

```diff
@@ -462,6 +462,11 @@ private void removeEntryOrLeaveTombstone() {
         focusedRegionMap.processVersionTag(newRegionEntry, event);
         if (doPart3) {
           internalRegion.generateAndSetVersionTag(event, newRegionEntry);
+          if (!event.getVersionTag().isGatewayTag()) {
+            // invoke listeners and inform clients
+            internalRegion.basicDestroyPart2(regionEntry, event, inTokenMode,
```

Comment:
Shouldn't this call be a few lines down after we have called newRegionEntry.makeTombstone on line 473?
Also shouldn't we be passing basicDestroyPart2 in this block of code "newRegionEntry" instead of "regionEntry"?